ui/static/ui/js/learningresources/learning_resource_panel.jsx

```diff
@@ -179,6 +194,15 @@ define("learning_resource_panel", ['react', 'jquery', 'lodash',
           thiz.setState({loaded: true});
         });
       },
+      componentWillUpdate: function(nextProps, nextState) {
```

Comment:
The logic here marks the dirty flag false and the logic in the callbacks above marks it true. Can we move this to a function like this:

    updateDirty: function() {
        var isDirty = !_.eq(this.state.description, this.state.descriptionOriginal) || !_.eq(this.state.vocabulariesAndTerms, this.state.vocabulariesAndTermsOriginal);
        this.props.markDirty(isDirty);
    }

Then just call `updateDirty` when there's any change to `vocabulariesAndTerms` or `description`